Index: Core/Banshee.ThickClient/Banshee.Gui.Widgets/UserJobTile.cs
===================================================================
--- Core/Banshee.ThickClient/Banshee.Gui.Widgets/UserJobTile.cs	(revision 3954)
+++ Core/Banshee.ThickClient/Banshee.Gui.Widgets/UserJobTile.cs	(working copy)
@@ -32,6 +32,7 @@
 
 using Banshee.Base;
 using Banshee.ServiceStack;
+using Hyena.Gui;
 
 namespace Banshee.Gui.Widgets
 {
@@ -177,6 +178,8 @@
                     status_label.Hide ();
                 } else {
                     status_label.Markup = String.Format ("<small>{0}</small>", GLib.Markup.EscapeText (job.Status));
+                    object tooltips = TooltipSetter.CreateHost ();
+                    TooltipSetter.Set (tooltips, status_label, job.Status);
                     status_label.Show ();
                 }
                 status = job.Status;
